mesonbuild/mdevenv.py

```diff
@@ -221,6 +229,11 @@ def run(options: argparse.Namespace) -> int:
             tmprc.flush()
             args.append("--rcfile")
             args.append(tmprc.name)
+        if macos_sip_enabled():
```

Comment:
Why not place `'DYLD_LIBRARY_PATH` in devenv and macos_sip_enabled()` here? Otherwise you might get a pointless warning.